woocommerce/woocommerce
# Set up "React Fast Refresh" in woocommerce-admin

### All Submissions:






### Changes proposed in this Pull Request:



Improve WooCommerce Admin's Development Workflow by using "React Fast Refresh":

- Add `start:hot` command in WCA to use React Fast Refresh
- Add necessary packages and update the webpack config.

Reference:
- https://github.com/WordPress/gutenberg/pull/28273



### How to test the changes in this Pull Request:




1. Run `cd ./plugins/woocommerce-admin`
2. Run `pnpm start`
3. Go to `WooCommerce > Home`
4. Edit any texts in the home screen such as ["You are almost there"](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/task-lists/progress-title/default-progress-title.tsx#L59)
5. Observe that it only refreshes the text when you fully reload the page.
6. Abort `pnpm start`
7. Run `pnpm start:hot`
8. Reload the page and repeat the step 3~4
9. Observe that it refreshes the text without a full page reload.



### Other information:




### FOR PR REVIEWER ONLY:
